### ISSUE TITLE
Unable to reference android .aar file using mvn deploy-file (was working before)

### ISSUE THREAD

**Issue by winbrother** [flutter/flutter#116962]
Hi there, 

I am running into issues with publishing .aar artifacts to apache maven (before it was working just fine)

In before, when I run 
```
flutter build aar
```

It will generate 3 folders: 
flutter_debug, flutter_profile and flutter_release

<img width="494" alt="Screen Shot 2022-12-13 at 3 40 38 PM" src="https://user-images.githubusercontent.com/18126130/207267793-635bede1-7e67-4c2c-8b4d-39cb0bd92c28.png">

From there, I can deploy them to mvn using the following command for debug build type
```
mvn deploy:deploy-file \
  -DrepositoryId=$GCPId \
  -DpomFile="$DirectoryPrefix\myPackageName\flutter_debug\\$VERSION\flutter_debug-$VERSION.pom" \
  -Dfile="$DirectoryPrefix\myPackageName\flutter_debug\\$VERSION\flutter_debug-$VERSION.aar" \
  -Durl=$GCPUrl
```

And I can deploy other build type accordingly

However, after migrating to flutter 3.3.9, running _flutter build aar_ will produce 4 folders:
flutter, flutter_debug, flutter_profile and flutter_release

I can still deploy .aar files for debug, profile and release using the above commands, but I can't do so for "flutter" one
I took a look inside, and see that there is flutter-1.1.0.pom file but no flutter-1.1.0.aar file

<img width="490" alt="Screen Shot 2022-12-13 at 3 42 03 PM" src="https://user-images.githubusercontent.com/18126130/207268063-2d7bd6f7-40d7-49f8-896e-924681c021a5.png">

At first I thought, the "flutter" folder is not relevant, but when I tried to integrate this .aar into my native android app, it complains that my "flutter_debug.aar" depends on "flutter" (in which I obviously unable to deploy to maven)

I tried each of these, and they produce the same error:

_debugImplementation("myPackageName:flutter:1.1.0")
_debugImplementation("myPackageName:flutter_debug:1.1.0")
_debugImplementation("myPackageName:flutter:1.1.0:debug")

From the native Android project, I ran gradle:dependencies and see this
```
+--- my.package.name:flutter:1.1.0 FAILED
```

I tried to do so to no avail. so I take a look at the flutter_debug-1.1.0.pom, and I found this

<img width="1092" alt="Screen Shot 2022-12-13 at 3 46 27 PM" src="https://user-images.githubusercontent.com/18126130/207268923-97e59502-cdf0-410e-9bb5-898b756f3adb.png">

I don't really understand what's going on here, but it seems I have to publish only the pom file for "flutter" with classifiers of the other 3 build types?

TLDR: before, "flutter build aar" will generate 3 build types of .aar, and I can deploy them to maven and consume them in my native Android app just fine. But now, it generate 4 build types, and I can't deploy the new one because they don't have corresponding .aar file with the .pom file. How can I deploy that file?

Currently I am unable to distribute new version to my fellow workers in the company.

Any help would be much appreciate.

**Comment by darshankawar** [comment]
Thanks for the report @winbrother 
What flutter version are you on ? Can you try the same on latest master (if you are not on it already) and check if you still get same behavior ?
Also, your case looks similar to https://github.com/flutter/flutter/issues/113417, so check underlying comments for your reference.
